### PR TITLE
refactor: improve app validation unit tests

### PR DESCRIPTION
**Pull request by jost-s**
### Summary
Move more arrange code to `TestCase`. Unit tests are more concise now.

### TODO:
- [ ] CHANGELOGs updated with appropriate info

### CODE REVIEW

File: crates/holochain/src/core/workflow/app_validation_workflow.rs [hunk 1]
```diff
@@ -49,8 +50,32 @@ mod unit_tests;
...
+
+impl ValidationDependencies {
+    pub fn new() -> Self {
+        Self {
+            missing_hashes: HashSet::new(),
+            hashes_missing_for_op: HashMap::new(),
+        }
+    }
+}
+
```
Comment (jost-s) [inline crates/holochain/src/core/workflow/app_validation_workflow.rs:75]: This snuck in from another PR and will be used for skipping op validation when dependencies are missing as well as ceasing to re-trigger the workflow as a whole when no further deps have been fetched in a certain period.

**Review by ThetaSinner — COMMENTED** [review]
Looks good. Slightly unsure about tying the test data to the test case but if it's sufficient test data to cover all the required tests then great

File: crates/holochain/src/core/workflow/app_validation_workflow.rs [hunk 2]
```diff
@@ -49,8 +50,32 @@ mod unit_tests;
...
 #[instrument(skip(
     workspace,
+    validation_dependencies,
     trigger_integration,
     conductor_handle,
     network,
```
[the remark this answers is not in the document]
Comment (jost-s) [reply crates/holochain/src/core/workflow/app_validation_workflow.rs:81]: It instruments the dna_hash, although in practice I've not found that useful as I'm looking at validations of different agents with the same DNA.

File: crates/holochain/src/core/workflow/app_validation_workflow.rs [hunk 3]
```diff
@@ -698,7 +722,10 @@ where
             // keep track of which dependencies are being fetched to
             // prevent multiple fetches of the same hash
             let validation_dependencies = validation_dependencies.clone();
-            let is_new_dependency = validation_dependencies.lock().insert(author.clone().into());
+            let is_new_dependency = validation_dependencies
```
[the remark this answers is not in the document]
Comment (jost-s) [reply crates/holochain/src/core/workflow/app_validation_workflow.rs:725]: cool

**Comment by jost-s** [comment]
> Looks good. Slightly unsure about tying the test data to the test case but if it's sufficient test data to cover all the required tests then great

I adapted all tests to use that test data and it's general enough but I was unsure myself and so I'll untie it again.

**Review by ThetaSinner — APPROVED** [review]
Like this much better, nice!

**Pull request merged**